hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4178,6 +4133,46 @@ protected void createLoaders() {
 		);
 	}
 
+	private UniqueEntityLoader getLoaderByLockMode(LockMode lockMode) {
+		if ( LockMode.NONE.equals( lockMode ) ) {
```

Comment:
It is common practice in Hibernate to use '==' to compare enum values for equality. You shall follow this practice here too.